docroot/modules/custom/sitenow_articles/sitenow_articles.module

```diff
@@ -231,6 +231,16 @@ function sitenow_articles_preprocess_node(&$variables) {
               }
             }
           }
+          // Check if preserve links checkbox is checked.
+          if ($node->hasField('field_article_preserve_links') && ($node->field_article_preserve_links->value == 0)) {
+            // If it is, do not display the broken links message.
+            foreach (array_reverse($variables['content']['_layout_builder'][1]['content'], TRUE) as $key => $block) {
```

Comment:
This code is based on the gallery jump link block, which uses array_reverse because it expects the gallery link to be last. I'm not sure if that is the case here, or how I can tell.

Reply:
Only really found out via xdebug, but I believe as blocks get added to the node display, they get added to the end of the list. I'd expect this one to be in the last space, and now the gallery link to be second to last.

It's a small optimization where reversing the array and then traversing it should be slightly more efficient if the blocks are at the end, but if/when more blocks get added, it'll be less and less of a difference.

Reply:
Oh cool, thanks for figuring it out! I'll leave the code as-is then.